python-lib/design/dku_tools.py

```diff
@@ -0,0 +1,97 @@
+import dataiku
+from dataiku.customrecipe import get_input_names_for_role, get_output_names_for_role
+from typing import List
+
+from design.constants import SizeDefinition, AttributionMethod, Parameters
+
+
+def get_input_output() -> tuple:
+    """Returns input and output datasets after sanity check
+
+    :raises: :class:`ValueError`: Missing input or output dataset(s)
+
+    :returns: input and output datasets
+    :rtype: tuple
+    """
+    input_names = get_input_names_for_role("user_list")
+    if len(input_names) == 0:
+        raise ValueError("No input dataset.")
+    output_names = get_output_names_for_role("groups")
+    if len(output_names) == 0:
+        raise ValueError("No output dataset.")
+
+    input_name = input_names[0]
+    input_dataset = dataiku.Dataset(input_name)
+    folder_ref = get_input_names_for_role('folder')
```

Comment:
I check this later because it is an optional input for the manual mode. So it raised an issue only when the size definition is the web app or folder mode.

Reply:
cf https://github.com/dataiku/dss-plugin-ab-testing/pull/1/commits/32502f58dd91d177dad5d154a484517b99faa1b0